frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -412,6 +412,30 @@ static ID parentFunctionId(Context* context, ID functionId) {
   return ID();
 }
 
+static void checkForParenlessMethodFieldRedefinition(Context* context,
+                                                     const Function* fn,
+                                                     Resolver& visitor) {
+
+  if (fn->isMethod() && fn->isParenless()) {
+    QualifiedType receiverType;
+    ID receiverId;
+    if (visitor.getMethodReceiver(&receiverType, &receiverId)) {
+      if (receiverType.type()) {
+        // use the type information, if it is present
+        if (auto ct = receiverType.type()->getCompositeType()) {
+          receiverId = ct->id();
+        }
+      }
+      if (!receiverId.isEmpty()) {
+        if (parsing::idContainsFieldWithName(context, receiverId, fn->name())) {
+          context->error(fn, "parenless proc redeclares the field '%s'",
+                         fn->name().c_str());
```

Comment:
It would be nice if this were a full-style Dyno error someday. Doesn't have to be in this PR, though.